### PR TITLE
Deregister v2

### PR DESCRIPTION
**Pull request by anoop44**

### CODE REVIEW

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/Deregister.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import android.view.ViewTreeObserver;
+
+/**
+ * @author Anoop S S
+ *         anoopvvs@gmail.com
+ *         on 28/02/2017
+ */
+
+public interface Deregister {
```
Comment (yshrsmz) [inline keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/Deregister.java]: Any reason you prefer `deregister` over `unregister`?
I think Android people are more familiar with `register - unregister` pair than `register - deregister`(such as `Context#registerReceiver` and `Context#unregisterReceiver`).

Also, class/interface name should be noun.

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
      * @param activity Activity
      * @param listener KeyboardVisibilityEventListener
      */
-    public static void setEventListener(final Activity activity,
+    public static Deregister setEventListener(final Activity activity,
```
Comment (yshrsmz) [inline keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.java]: My bad, to meet the naming consistency, this should be renamed to `registerEventListener`

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/Deregister.java (hunk 1 above)
Comment (anoop44) [inline keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/Deregister.java]: Even I was using unregister till I read this http://english.stackexchange.com/questions/25931/unregister-vs-deregister

Comment (anoop44) [inline keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/Deregister.java]: I think we will go with the common usgea 'unregister'. Will update it with required changes

**Comment by yshrsmz** [comment]
thanks! will release in a few days after some cleanup :)

**Pull request merged**